dist/extensions/hydra.md

```diff
@@ -106,6 +106,66 @@ The crucial detail is matching a template's `hydra:property` with another proper
 > [!EXAMPLE]
 > See an example shape in the [Playground][template]
 
+#### Selecting template value using a complex path
+
+Consider the resource below.
+
+```turtle
+@prefix schema: <http://schema.org/> .
+
+<>
+  a schema:Vehicle ;
+  schema:engine [
+    schema:enginePower [
+      schema:value "120" ;
+      schema:unit "PS" ;        
+    ] ;
+  ] ;
+  schema:fuelType "DIESEL" ;
+.
+```
+
+To construct a URI using the values of `schema:fuelType` and `schema:engine/schema:enginePower/schema:value` requires that
+property paths are used. That is not possible with `hydra:mapping/hydra:property` because it only allows a single RDF 
+predicate.
+
+`@hydrofoil/shaperone-hydra` supports an additional `sh:path` property on the template mappings so that values are not 
+limited only to the given focus node.
+
+```turtle
+@prefix schema: <http://schema.org/> .
+@prefix ex: <http://example.org/> .
+@prefix hydra: <http://www.w3.org/ns/hydra/core#> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+
+<>
+  sh:property [
+    sh:path ex:related ;
+    hydra:search [
+      a hydra:IriTemplate ;
+      hydra:template "https://example.com/cars{?power,fuel}" ;
+      hydra:mapping [
+        hydra:property schema:fuelType ;
+        hydra:variable "fuel" ;
+      ],  [
+        hydra:property schema:enginePower ;
```

Comment:
It's clear the spirit of sh:path, and seems very flexible!

However, In this case, isn't hydra:property here redundant? Do I get the example well?

Reply:
They serve a different purpose. `hydra:property` is still needed to actually needed to build the final URI. It is used to connect the URI Template variable with the object in RDF.

In the background, a graph is constructed like below and only from it the URL is constructed.

```turtle
[ 
  schema:enginePower "120" ;
  schema:fuelType "DIESEL" ;
] .
```

The change here is to find the value `"120"` in another node by following the path instead of `hydra:property`

Reply:
Then I did not get the example. 

Isn't sh:path defining the path sequence: ( schema:engine schema:enginePower schema:Value), where schema:Value is the object to consider?, (thus, schema:enginePower the implicit property?)

Reply:
You're right. The URI template mappings define which RDF values (`hydra:property`) and used in the template (`hydra:variable`). The input for this operation is a graph pointer.

`sh:path` is necessary here to create a pointer which gathers values from multiple resource in the graph because there is no one node which has both `enginePower` and `fuleType` directly

Reply:
I'm afraid I still don't understand why ` hydra:property schema:enginePower ;` is necessary. 

Perhaps I don't understand what `https://example.com/cars?power=120,fuel=DIESEL`. returns.

In this example, `power` is the variable to search, bonded to schema:enginePower, and the search results will be schema:value objects ?

Reply:
What it returns is not important. I'm "guessing" cars :). 

Coming back to what I said above, the input to filling in a template is a graph pointer. Prior to this change it would implicitly be the current focus node. In this example it's the `<> a schema:Vehicle` resource. The mapping would look like this, taking the values from that resource node:

| template variable | `hydra:property` | value |
| -- | -- | -- |
| `fuel` | `schema:fuelType` | `"DIESEL"` |
| `power` | `schema:enginePower` | `blank node` |

That does not provide the expected result. You need a new way to access the value for `power`:


| template variable | `hydra:property` | `sh:path` | value |
| -- | -- | -- |
| `fuel` | `schema:fuelType` | | `"DIESEL"` |
| `power` | `schema:enginePower` |  `schema:engine/schema:enginePower/schema:value` | `"120"` |

Reply:
☕ ☕ ☕ ☕ 

What I mentioned, above, maybe incomplete. The `hydra:template` is defined in the Hydra spec. It maps a resources with properties to a template and also the other way. Let's simplify only to a single variable and follow the full flow.

Given a simple template

```turtle
[ 
  hydra:template "/cars{?fuel}" ; 
  hydra:mapping [ hydra:variable "fuel" ; hydra:property schema:fuelType ]
]
```

when you add input a focus node `[ schema:fuelType "DIESEL" ]`, you will get `/cars?fuel=DIESEL`. The server does the opposite operation, and extracts the exact same graph from the request parameters.

Now, when looking at the engine power variable. The template would look similar

```turtle
[ 
  hydra:template "/cars{?power}" ; 
  hydra:mapping [ hydra:variable "power" ; hydra:property schema:enginePower ]
]
```

And given a request for `/cars?power=120`, the server would reconstruct that as `[ schema:enginePower "120" ]`, likely example to do query filtering.

The problem, solved here, is that in the form the graph you would have the deep structure of

```turtle
[
  schema:engine [ schema:enginePower [ schema:value "120" ] ] ]
]
```

Hence, the additional `sh:path` lets the form select the right values to bind the template. But `hydra:property` is still necessary for both constructing and deconstructing the URL.

Reply:
In your example, why would one want to deconstruct the URL?
If `/cars?power=120` return cars, shouldn't each car return (or not) their corresponding `schema:enginePower`?


But going backward, I need some information to better understand Hydra (which I completely forgot after many years)

I'm looking at example 29 at: https://www.hydra-cg.com/spec/latest/core/#example-29-supported-property-data-source

In the example, 

```turtle
{
  "@context": "http://www.w3.org/ns/hydra/context.jsonld",
  "@type": "ApiDocumentation",
  "supportedClass": {
    "@id": "schema:Event",
    "supportedProperty": {
      "property": "schema:actor",
      "search": {
        "template": "/api/user{?search}",
        "mapping": {
          "variable": "search",
          "property": "freetextQuery"
        }
      }****
    }
  },
  ...
}
```

Given the example, is this represented when using SHACL as:

```turtle
@prefix schema: <http://schema.org/> .
@prefix ex: <http://example.org/> .
@prefix hydra: <http://www.w3.org/ns/hydra/core#> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
<>
  sh:property [
    sh:path schema:actor ;
    hydra:search [
	...
``` ?

I'm a bit confused about what to use SHACL in the responses of Hydra. (right now, I don't necessarily see Hydra as related to SHACL, since I haven't seen examples of it)

Where can I read more about it? Is that necessary?